### PR TITLE
add :m[ove]

### PR DESCRIPTION
**Pull request by zhanyi22333**
**What this PR does / why we need it**:
Add "move" excommand.
For excommand like : a,bmc  (a,b,c are line numbers), it works as follow:
1. Copy text in range [a,b].
2. Paste text below c, in range[c+1,c+1+(b-a)].
3. delete text in  range [a,b].

Futhermore, I just ban the situation that range [a,b] and range[c+1,c+1+(b-a)], which will lead to wrong performance and has no practical use. I also ban the situation that move [a,b] to [a,b], in which case the "move" excommand just do useless work.

**Special notes for your reviewer**:
I'm just a newbie in PR.Let me know if there is something not in convention.

### CODE REVIEW

File: src/cmd_line/commands/move.ts
Comment (J-Fields) [inline src/cmd_line/commands/move.ts:1]: Can you write a few test cases for `:move` please?

**Review by J-Fields — CHANGES_REQUESTED** [review]

File: src/cmd_line/commands/move.ts
Comment (zhanyi22333) [reply src/cmd_line/commands/move.ts:1]: Ok.I'll work on this.
By the way,ls there any doc for writing test or a good sample test script?

Comment (J-Fields) [reply src/cmd_line/commands/move.ts:1]: `sort.test.ts` is a decent example

**Comment by zhanyi22333** [comment]
I've add a test script for m[ove].During the process, I compare its behavior with orignal vim and find some misunderstanding. So I change the workflow and also change the ban situation.
Now, the performannce is finetuned like this:
For excommand like : a,bmc (a,b,c are line numbers), it works as follow:

1. Copy text in range [a,b].
2. Find target position  by c and decide cursor position.
3. Delete text in range [a,b].
4. Paste text to target positon and move cursor to cursor position.

Ban the situtations that  move a range into itself and the situations that move to adress just right abow or below the range.

All test scripts are passed on my pc. If there are some problems with the added test script, let me know. Thanks.

**Comment by zhanyi22333** [comment]
There are two tests failed when build on windows-lastest. I've re-test on my pc and can not reproduce them. And the failed tests seem no relationship with m[ove]. I feel confused. Should I do something for the failed tests?

**Comment by J-Fields** [comment]
> Should I do something for the failed tests?

Nah don't worry about it; a few tests are just flaky

**Review by J-Fields — APPROVED** [review]
Thanks!

**Comment by Miigon** [comment]
Thank god this is implemented!

**Pull request merged**